tests/test_sft_trainer.py

```diff
@@ -858,6 +861,102 @@ def test_run_causallm_ft_and_inference_with_multiple_dataset(
         assert "### Text: @NortonSupport Thanks much.\n\n### Label:" in output_inference
 
 
+@pytest.mark.parametrize(
+    "dataset_path",
+    [CHAT_DATA_SINGLE_TURN, CHAT_DATA_MULTI_TURN],
+)
+def test_run_chat_style_ft(dataset_path):
+    """Check if we can perform an e2e run with chat template and multi turn chat training."""
+    with tempfile.TemporaryDirectory() as tempdir:
+
+        data_args = copy.deepcopy(DATA_ARGS)
+        data_args.training_data_path = dataset_path
+        data_args.chat_template = "{% for message in messages['messages'] %}\
+            {% if message['role'] == 'user' %}{{ '<|user|>\n' + message['content'] + eos_token }}\
+            {% elif message['role'] == 'system' %}{{ '<|system|>\n' + message['content'] + eos_token }}\
+            {% elif message['role'] == 'assistant' %}{{ '<|assistant|>\n'  + message['content'] + eos_token }}\
+            {% endif %}\
+            {% if loop.last and add_generation_prompt %}{{ '<|assistant|>' }}\
+            {% endif %}\
+            {% endfor %}"
+        data_args.response_template = "<|assistant|>"
+        data_args.instruction_template = "<|user|>"
+
+        model_args = copy.deepcopy(MODEL_ARGS)
+        model_args.tokenizer_name_or_path = CUSTOM_TOKENIZER_TINYLLAMA
+
+        train_args = copy.deepcopy(TRAIN_ARGS)
+        train_args.output_dir = tempdir
+
+        sft_trainer.train(model_args, data_args, train_args)
+
+        # validate full ft configs
+        _validate_training(tempdir)
+
```

Comment:
Shall we extend these test cases a bit add the inference part as well testing for some indication of success in quality? Like we did for older ones https://github.com/foundation-model-stack/fms-hf-tuning/pull/415/files#diff-406c3c4d02c1ff158009b04d26a6b2d357f05d11e8c35c3f2ddfc1b54649a022R854-R861?

I think we can ignore this extra layer of validation for the data config test case. Thanks.

Reply:
I was not sure if thats needed but just to be safe I added for both as other tests had for all of them